op-service/dial/active_rollup_provider.go

```diff
@@ -106,37 +100,46 @@ func (p *ActiveL2RollupProvider) shouldCheck() bool {
 	return time.Now().After(p.activeTimeout)
 }
 
-func (p *ActiveL2RollupProvider) ensureClientInitialized(ctx context.Context) error {
-	if p.currentRollupClient != nil {
-		return nil
-	}
-	return p.dialNextSequencer(ctx)
-}
-
 func (p *ActiveL2RollupProvider) findActiveEndpoints(ctx context.Context) error {
-	for range p.rollupUrls {
-		active, err := p.checkCurrentSequencer(ctx)
-		if errors.Is(err, errSeqUnset) {
-			log.Debug("Current sequencer unset.")
-		} else if ep := p.rollupUrls[p.rollupIndex]; err != nil {
-			p.log.Warn("Error querying active sequencer, closing connection and trying next.", "err", err, "index", p.rollupIndex, "url", ep)
+	startIdx := p.rollupIndex
+	var errs error
+	for offset := range p.rollupUrls {
+		idx := (startIdx + offset) % p.numEndpoints()
+		if offset != 0 || p.currentRollupClient == nil {
+			if err := p.dialSequencer(ctx, idx); err != nil {
+				errs = errors.Join(errs, err)
+				p.log.Warn("Error dialing next sequencer.", "err", err, "index", p.rollupIndex)
+				continue
+			}
+		}
+
+		ep := p.rollupUrls[idx]
+		if active, err := p.checkCurrentSequencer(ctx); err != nil {
+			errs = errors.Join(errs, err)
+			p.log.Warn("Error querying active sequencer, trying next.", "err", err, "index", idx, "url", ep)
 		} else if active {
-			p.log.Debug("Current sequencer active.", "index", p.rollupIndex, "url", ep)
+			if offset == 0 {
+				p.log.Debug("Current sequencer active.", "index", idx, "url", ep)
+			} else {
+				p.log.Info("Found new active sequencer.", "index", idx, "url", ep)
+			}
 			return nil
 		} else {
-			p.log.Info("Current sequencer inactive, closing connection and trying next.", "index", p.rollupIndex, "url", ep)
-		}
-		if err := p.dialNextSequencer(ctx); err != nil {
-			p.log.Warn("Error dialing next sequencer.", "err", err, "index", p.rollupIndex)
+			p.log.Info("Sequencer inactive, trying next.", "index", idx, "url", ep)
 		}
 	}
-	return fmt.Errorf("failed to find an active sequencer, tried following urls: %v", p.rollupUrls)
+
+	// TODO: We exhausted the list of sequencers, close and remove latest client?
+	//   It definitely makes all the existing tests pass, but maybe we remove
+	//   this and adjust the tests, if it makes sense.
```

Comment:
```suggestion
```
I think it makes sense to `Close()` and clean up the rollup client that isn't being used, no?

Reply:
We discussed that we should try to remove this `nil`-ification altogether. It might save re-dials of an inactive, but online sequencer.